packages/web/src/components/upload/TrackPreviewNew.tsx

```diff
@@ -0,0 +1,83 @@
+import { useMemo } from 'react'
+
+import { HarmonyButton, HarmonyButtonType, IconTrash } from '@audius/stems'
+import numeral from 'numeral'
+
+import iconFileAiff from 'assets/img/iconFileAiff.svg'
+import iconFileFlac from 'assets/img/iconFileFlac.svg'
+import iconFileMp3 from 'assets/img/iconFileMp3.svg'
+import iconFileOgg from 'assets/img/iconFileOgg.svg'
+import iconFileUnknown from 'assets/img/iconFileUnknown.svg'
+import iconFileWav from 'assets/img/iconFileWav.svg'
+import { Text } from 'components/typography'
+
+import styles from './TrackPreview.module.css'
+
+const fileTypeIcon = (type: string) => {
+  switch (type) {
+    case 'audio/mpeg':
+    case 'audio/mp3':
+      return iconFileMp3
+    case 'audio/aiff':
+      return iconFileAiff
+    case 'audio/flac':
+      return iconFileFlac
+    case 'audio/ogg':
+      return iconFileOgg
+    case 'audio/wav':
+      return iconFileWav
+    default:
+      return iconFileUnknown
+  }
+}
+
+type TrackPreviewProps = {
+  fileType: string
+  trackTitle: string
+  fileSize: number
+  index: number
+  displayIndex: boolean
+  onRemove: () => void
+}
+
+export const TrackPreviewNew = (props: TrackPreviewProps) => {
+  const {
+    displayIndex = false,
+    index,
+    fileType = 'audio/mp3',
+    trackTitle = 'Untitled',
+    fileSize,
+    onRemove
+  } = props
+
+  const imgSrc = useMemo(() => fileTypeIcon(fileType), [fileType])
```

Comment:
since this is a simple if/else branch function, its more performant to just let the computation happen. Typically only when the computation itself is expensive, or what's being returned is not a primitive value, like an array or object, is useMemo necessary

Reply:
can we also name this better, like fileTypeSrc? also we can probably just put this inline in the img component, too